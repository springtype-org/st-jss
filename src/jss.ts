import { FontFaceProperties, KeyFrameProperties } from 'springtype-types';
import { API } from './interface/API';
import { ClassProperty } from './interface/ClassProperties';
import { makeStyles } from './makeStyles';
import { makeFont } from './makeFont';
import { makeAnimation } from './makeAnimation';

export const jss: API = {
  defaultUnit: 'px',
  classNamePrefix: 'st',
  instanceCounter: 0,
  makeAnimation: (animationName: string, style: Partial<KeyFrameProperties>) =>
    makeAnimation(animationName, style, jss),
  makeFont: (style: FontFaceProperties) => makeFont(style),
  makeStyles: <T extends string = string>(classProperty: ClassProperty<T>) => makeStyles<T>(classProperty, jss),
};
