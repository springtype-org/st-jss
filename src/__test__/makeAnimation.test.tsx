import { jss } from '../jss';

describe('makeAnimation', () => {
  it('check style output', () => {
    const [animationName, animationStyle] = jss.makeAnimation('animation', {
      '0%': { top: 0, left: 0, background: 'red' },
      '25%': { top: 0, left: 100, background: 'blue' },
      '50%': { top: 100, left: 100, background: 'yellow' },
      '75%': { top: 100, left: 0, background: 'green' },
      '100%': { top: 0, left: 0, background: 'red' },
    });
    expect(animationName).toMatchSnapshot();
    const { children: childrenOfAnimation, ...otherPropsOfAnimation } = animationStyle;
    expect(childrenOfAnimation[0]).toMatchSnapshot();
    expect(otherPropsOfAnimation).toMatchSnapshot();
  });
});
