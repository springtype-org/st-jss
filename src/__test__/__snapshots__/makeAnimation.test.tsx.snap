// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeAnimation check style output 1`] = `"st-animation-1"`;

exports[`makeAnimation check style output 2`] = `
"@keyframes st-animation-1 {
  0% {
    top: 0px;
    left: 0px;
    background: red;
  }
  25% {
    top: 0px;
    left: 100px;
    background: blue;
  }
  50% {
    top: 100px;
    left: 100px;
    background: yellow;
  }
  75% {
    top: 100px;
    left: 0px;
    background: green;
  }
  100% {
    top: 0px;
    left: 0px;
    background: red;
  }
}"
`;

exports[`makeAnimation check style output 3`] = `
Object {
  "attributes": Object {
    "type": "text/css",
  },
  "type": "style",
}
`;
